odl/solvers/functional/default_functionals.py

```diff
@@ -0,0 +1,683 @@
+    """The constant functional.
+
+    This functional maps all elements in the domain to a given, constant value.
+    """
+    def __init__(self, domain, constant):
+        """Initialize a ConstantFunctional instance.
+
+        Parameters
+        ----------
+        domain : `LinearSpace`
+            The space of elements which the functional is acting on.
+        constant : element in `domain.field`
+            The constant value of the functional
+        """
+
+        super().__init__(domain=domain, linear=True, convex=True,
+                         concave=True, smooth=True, grad_lipschitz=0)
+
+        if constant not in self.range:
+            raise TypeError('constant {} not in the range {}.'
+                            ''.format(constant, self.range))
+
+        self._constant = constant
+
+    @property
+    def constant(self):
+        return self._constant
+
+    def _call(self, x):
+        """Applies the functional to the given point.
+
+        Parameters
+        ----------
+        x : `LinearSpaceVector`
+            Element in the domain of the functional.
+
+        Returns
+        -------
+        `self(x)` : `float`
+            Evaluation of the functional, which is a constant.
+        """
+        return self._constant
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional.
+
+        Notes
+        -----
+        The operator that corresponds to the mapping
+
+        .. math::
+
+            x \\to \\nabla f(x)
+
+        where :math:`\\nabla f(x)` is the element used to evaluated
+        derivatives in a direction :math:`d` by
+        :math:`\\langle \\nabla f(x), d \\rangle`.
+        """
+        return ZeroOperator(self.domain)
+
+    def proximal(self, sigma=1.0):
+        """Return the proximal operator of the constant functional.
+
+        The proximal operator for the constant functional is the identity
+        operator.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Regularization parameter of the proximal operator
+
+        Returns
+        -------
+        out : Operator
+            Domain and range equal to domain of functional
+        """
+
+        return IdentityOperator(self.domain)
+
+    @property
+    def conjugate_functional(self):
+        """Convex conjugate functional of the constant functional.
+
+        This functional ``f^*``  is such that it maps the zero element to zero,
+        the rest to infinity.
+        """
+        functional = self
+
+        class ConstantFunctionalConjugateFunctional(Functional):
+            """The convex conjugate functional to the constant functional."""
+
+            def __init__(self):
+                """Initialize a ConstantFunctionalConjugateFunctional
+                instance.
+                """
+                super().__init__(functional.domain, linear=False, convex=True)
+                self.zero_element = self.domain.zero()
```

Comment:
Is this useful for speed, since call is going to check against this element every time?